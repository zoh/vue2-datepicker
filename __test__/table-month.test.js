import { mount } from '@vue/test-utils';
import TableMonth from '../src/calendar/table-month.vue';

let wrapper;

afterEach(() => {
  wrapper.destroy();
});

describe('TableMonth', () => {
  it('correct render', () => {
    wrapper = mount(TableMonth, {
      propsData: {
        getClasses: month => {
          if (month === 9) {
            return 'active';
          }
          return '';
        },
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
