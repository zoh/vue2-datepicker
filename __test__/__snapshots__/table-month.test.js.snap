// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableMonth correct render 1`] = `
<table
  class="mx-table mx-table-month"
  getclasses="month => {
          if (month === 9) {
            return 'active';
          }

          return '';
        }"
>
  <tr>
    <td
      class="cell"
      data-month="0"
    >
      <div>
        Jan
      </div>
    </td>
    <td
      class="cell"
      data-month="1"
    >
      <div>
        Feb
      </div>
    </td>
    <td
      class="cell"
      data-month="2"
    >
      <div>
        Mar
      </div>
    </td>
  </tr>
  <tr>
    <td
      class="cell"
      data-month="3"
    >
      <div>
        Apr
      </div>
    </td>
    <td
      class="cell"
      data-month="4"
    >
      <div>
        May
      </div>
    </td>
    <td
      class="cell"
      data-month="5"
    >
      <div>
        Jun
      </div>
    </td>
  </tr>
  <tr>
    <td
      class="cell"
      data-month="6"
    >
      <div>
        Jul
      </div>
    </td>
    <td
      class="cell"
      data-month="7"
    >
      <div>
        Aug
      </div>
    </td>
    <td
      class="cell"
      data-month="8"
    >
      <div>
        Sep
      </div>
    </td>
  </tr>
  <tr>
    <td
      class="cell"
      data-month="9"
    >
      <div>
        Oct
      </div>
    </td>
    <td
      class="cell"
      data-month="10"
    >
      <div>
        Nov
      </div>
    </td>
    <td
      class="cell"
      data-month="11"
    >
      <div>
        Dec
      </div>
    </td>
  </tr>
</table>
`;
